src/components/registration/LogIn.js

```diff
@@ -26,53 +36,91 @@ class LogIn extends Component {
         localStorage.setItem('app-token', response.data.token);
-        // this.props.history.push("/dashboard");
-        this.props.handleLogin();
+        handleLogin();
       })
       .catch(error => {
-        try {
-          // Handles errors that are not HTTP specific
-          console.error(error);
-          this.setState({ showRegistrationFailure: true });
-          if (!error.status) {
-            console.error('A network error has occured.');
-          } else if (error.response.status === 400) {
-            console.error('Bad Request');
-          } else if (error.response.status === 500) {
-            console.error('Something bad happended on the server.');
-          } else {
-            console.error('An unknown error has occurred');
-          }
-        } catch (ex) {
-          Promise.reject(ex);
+        if (error.message === 'Network Error') {
+          this.setState({
+            HTTPErrorMessage:
+              'A network error has occurred while contacting our servers...',
+          });
+        } else if (error.response.status === 400) {
+          this.setState({ HTTPErrorMessage: `${error.response.data}` });
+        } else if (error.response.status === 500) {
+          this.setState({
+            HTTPErrorMessage: `Something went wrong on our side. Please try again at a later time.\n Error: ${error.response.data}`,
+          });
         }
       });
   };
 
-  render() {
+  handleEmailChange = evt => {
+    this.setState({ email: evt.target.value });
+  };
+
+  handlePasswordChange = evt => {
+    this.setState({ password: evt.target.value });
+  };
+
+  validateData = () => {
     const { email, password } = this.state;
+    let setErrors = { email: false, password: false };
+    let atLeastOneFailed = false;
+    // Validating Email
+    if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email)) {
+      setErrors.email = true;
+      atLeastOneFailed = true;
+    }
+    // Validating password
+    if (password.length === 0 || password === undefined) {
+      setErrors.password = true;
+      atLeastOneFailed = true;
+    }
+    // Setting error flags in state
+    if (!atLeastOneFailed) {
+      this.LogInHandler();
+    } else {
+      this.setState({ errors: setErrors });
+    }
+  };
+
+  render() {
+    const { email, password, errors, HTTPErrorMessage } = this.state;
     const { changeToSignup } = this.props;
     return (
       <div id="login-container">
         <h1 id="title">Login</h1>
         <form id="login-form">
           <h3>Email</h3>
           <input
-            type="email"
+            className={errors.email ? 'formError' : null}
+            type="text"
             value={email}
-            onChange={event => this.setState({ email: event.target.value })}
+            onChange={this.handleEmailChange}
+            placeholder=""
           />
+          {errors.email ? (
+            <p className="ErrorMessage">{this.state.ValidationErrorMessages.badEmail}</p>
```

Comment:
Update this line once you take it out of state
```suggestion
            <p className="ErrorMessage">{this.validation_error_messages.bad_email}</p>
```